internal/ec_errors/ec_error.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2022 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package ec_errors
+
+type Error struct {
+	code       string
+	message    string
+	cause      string
+	exitStatus int
```

Comment:
My main concern with this is scope. A helper function nested in the code base is probably not the right place to decide if a certain error means the process should exit, let alone which status code.

I can see this being used directly in the code under `cmd/*.go`. But using this approach in every function that can return an error seems unnecessarily cumbersome.

Reply:
Most cases we propagate the same error up the stack. I don't see a way around this. Also it is essential to have many different error codes to know where the error originated. Otherwise we'll have very difficult time troubleshooting. When we "handle" an error, we need to generate a new error constant here and at that point we define the exit status as well.

Reply:
It's an approach worth trying out :+1: